api/src/main/java/org/openmrs/module/fhir/api/strategies/condition/ConditionStrategy.java

```
@@ -0,0 +1,58 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+
+package org.openmrs.module.fhir.api.strategies.condition;
+
+import org.hl7.fhir.dstu3.model.Condition;
+import org.openmrs.module.fhir.api.util.ContextUtil;
+import org.springframework.stereotype.Component;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.util.List;
+
+@Component("DefaultConditionStrategy")
+public class ConditionStrategy implements GenericConditionStrategy {
+
+	/**
+	 * @see org.openmrs.module.fhir.api.strategies.condition.GenericConditionStrategy#getConditionById(String)
+	 */
+	@Override
+	public Condition getConditionById(String uuid) {
+		return ContextUtil.getConditionHelper().getCondition(uuid);
+	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.strategies.condition.GenericConditionStrategy#searchConditionById(String)
+	 */
+	@Override
+	public List<Condition> searchConditionById(String uuid) throws NotImplementedException {
+		return null;
+	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.strategies.condition.GenericConditionStrategy#searchConditionByName(String)
+	 */
+	@Override
+	public List<Condition> searchConditionByName(String name) throws NotImplementedException {
+		return null;
```

I mean throwing NotImplementedException instead of returning null